### PR TITLE
Update CHANGELOG.md for recent changes, in particular source breaks

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
I'll pick this over to the swift-5.0-branch once I get a review on the content.

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,12 +23,40 @@ CHANGELOG
...
+  flag are `4`, `4.2`, and `5`.
+
+* [SE-0228][]:
+
+  String interpolation has been overhauled to improve its performance,
+  clarity, and efficiency.
+
+  Note that the old `_ExpressibleByStringInterpolation` protocol has been
+  removed; any code making use of this protocol will need to be updated
+  for the new design.
```
Comment (jrose-apple) [inline CHANGELOG.md]: Should we mention `#if compiler` as a useful thing here? (That's how I tested an internal Apple project that was using `_ExpressibleByStringInterpolation`.)

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +68,56 @@ Swift 5.0
...
+  Protocols can now constrain their conforming types to those that subclasses a
+  given class. Two equivalent forms are supported:
+
+  ```swift
+  protocol MyView : UIView { ... }
+  protocol MyView where Self : UIView { ... }
+  ```
+
+  Note that Swift 4.2 accepted the second form, but it was not fully implemented
+  and could crash.
```
Comment (jrose-apple) [inline CHANGELOG.md]: "could cause crashes at run time", maybe?

File: CHANGELOG.md [hunk 3]
```diff
@@ -129,16 +207,24 @@ Swift 5.0
...
+* `unowned` and `unowned(unsafe)` variables now support Optional types.
+
+* Designated initializers with variadic parameters are now correctly inherited
+  in subclasses.
+
+* Extensions of concrete subclasses of generic classes can now contain
+  `@objc` members.
+
+* Complex recursive type definitions involving classes and generics that would
+  previously deadlock the runtime when realizing metadata are now supported.
```
Comment (jrose-apple) [inline CHANGELOG.md]: I don't think "realizing metadata" is a user-facing term. Maybe just "initializing type metadata"?

**Review by jrose-apple — APPROVED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**